File: hashmap/firt-unique-character-in-string.py
class Solution:
  def firstUniqChar(self, s: str) -> int:
    'l': [0,1]
    'e': [1,13]
    d = {}
    for idx, ch in enumerate(s): 
        if not d.get(ch)
        d[ch] = [idx, 1]
        else:
          d[ch][1] += 1
          
          for ch, val in d.items():
            if val[1] == 1:
              return val[0]
            
            return -1 

File: array/hashmap/firt-unique-character-in-string.js
class Solution {
  firstUniqChar(s) {
    const d = new Map();

    for (let idx = 0; idx < s.length; idx++) {
      const ch = s[idx];
      if (!d.has(ch)) {
        d.set(ch, [idx, 1]);
      } else {
        const val = d.get(ch);
        val[1] += 1;
        d.set(ch, val);
      }
    }

    for (const [ch, val] of d.entries()) {
      if (val[1] === 1) {
        return val[0]; 
      }
    }

    return -1; 
  }
}

const sol = new Solution();
console.log(sol.firstUniqChar("leetcode")); 
console.log(sol.firstUniqChar("loveleetcode")); 
console.log(sol.firstUniqChar("aabb")); 